lib/shared_print/replace_record.rb

```diff
@@ -0,0 +1,80 @@
+module SharedPrint
+  class ReplaceRecord
+    attr_reader :existing, :replacement
+    def initialize(existing: nil, replacement: nil)
+      @existing = existing
+      @replacement = replacement
+      validate
+    end
+
+    def validate
+      raise ArgumentError, "Missing existing commitment" if @existing.nil?
+      raise ArgumentError, "Missing replacement commitment" if @replacement.nil?
+
+      # The replacement does not have to be a commitment that exists in the db.
+      # It may need to autovivify upon replacing an old one, when reading from file.
+      # Thus we may need to turn a Hash read from a file into a Clusterable::Commitment.
+      if @replacement.instance_of?(Hash)
+        @replacement = Clusterable::Commitment.new(@replacement)
+        Clustering::ClusterCommitment.new(@replacement).cluster.tap(&:save)
+      end
+
+      unless @existing.instance_of?(Clusterable::Commitment)
+        raise ArgumentError, "Existing commitment must be a Clusterable::Commitment"
+      end
+
+      unless @replacement.instance_of?(Clusterable::Commitment)
+        raise ArgumentError, "Replacement must be a Clusterable::Commitment"
+      end
+
+      unless matches.size == 1
+        raise IndexError, "Existing matches must be 1 (is #{matches.size})"
+      end
+    end
+
+    # Replace :existing with :replacement.
+    def apply
+      # Sadly (?) this makes it so that all replaced commitments are deprecated with
+      # the same status "E". Even if previously deprecated with another status.
+      # Status "R" for Replaced was rejected because it would be meaningless.
+      matches.first.deprecate(status: "E", replacement: @replacement)
+      matches.first.save
+    end
+
+    # Original implementation that creates duplicate records with the same _ids in Mongo.
+    def apply_broken
```

Comment:
There a reason for this, or just forgot to rm?

Reply:
This was intentionally left as a nod to @aelkiss  saying "Worth having a failing test for this case." on https://tools.lib.umich.edu/jira/browse/HT-3351, but I forgot to actually put the test in.

Reply:
Test added.